datafusion/core/Cargo.toml

```diff
@@ -47,6 +47,7 @@ backtrace = ["datafusion-common/backtrace"]
 compression = ["xz2", "bzip2", "flate2", "zstd", "async-compression", "tokio-util"]
 crypto_expressions = ["datafusion-functions/crypto_expressions"]
 datetime_expressions = ["datafusion-functions/datetime_expressions"]
+dirs = ["dep:dirs"]
```

Comment:
I'm not sure which way is better to make `dirs` be optional. So I just create a new feature for it.